### ISSUE TITLE
Support for classes

### ISSUE THREAD

**Issue by gmistick** [evilsocket/smali_emulator#13]
The emulator currently support running files that contain the content of one single method.

Maybe its a good idea to start discussion about supporting classes, subclasses, and dynamic object-mappings created for each read file.

**Comment by h3ku** [comment]
I think it supports one entire class, but not support calls to other classes created by you.

**Comment by gmistick** [comment]
Well, the run function just runs the first function.. So now really..
Also static fields are not working.

Etc

**Comment by h3ku** [comment]
Yep, my fault, I make a test to try it, I see it run the constructor so I assume it works, sorry xD

**Comment by drake-mer** [comment]
I think the focus should be put on supporting static classes and attributes only for now.

I have some work in progress toward this direction.

Class resolution across multiple files could indeed be implemented but this would be part of a larger effort but from what I've seen this is not really useful for deobfuscation purpose. Plus the degree of complexity is just ++.

**Comment by GM12Tick** [comment]
Well, it would be good to have a discussion.
The real question is, do you think that smali emulator can replace the android emulator ? Be the python version for running apks :)

**Comment by drake-mer** [comment]
It would be very cool as an asymptotic goal.
Method resolution in an internal class is also a cool goal.

**Comment by drake-mer** [comment]
I have worked a bit toward class emulation and from what I did I can tell it is quite doable. I still need some thought for where to put the methods (in the vm, in the emulator, circular reference ?) but I can do it.
